src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -859,6 +859,16 @@ public bool ValidateTransaction(Transaction transaction, bool checkSignature = f
                 if (!this.TransactionHasValidUTXOs(transaction, coins))
                     return false;
 
+                // Verify that there are no earlier unspent UTXOs.
```

Comment:
Maybe create a log entry / console warning if this happens as we don't expect t to.